lightclient/src/background.rs

```diff
@@ -446,3 +473,23 @@ impl std::str::FromStr for RpcResponse {
         })
     }
 }
+
+/// Generate unique IDs for RPC calls that are distinct for each chain.
+#[derive(Default)]
+struct UniqueChainID {
```

Comment:
As said previously I think this can be removed but your choice

Reply:
Yep, this turned out to be quite small in the end. Worth refactoring it into the main background struct, thanks for the feedback here and above 🙏